lib/torrent.js

```diff
@@ -1541,16 +1541,21 @@ Torrent.prototype._request = function (wire, index, hotswap) {
     })
   })
 
+  function onDone () {
+    self.discovery.complete()
+  }
+
   function onUpdateTick () {
     process.nextTick(function () { self._update() })
   }
 
   return true
 }
 
-Torrent.prototype._checkDone = function () {
+Torrent.prototype._checkDone = function (cb) {
```

Comment:
No need for a callback. This method is entirely sync. Let's just return `true` if the entire torrent is done, `false` otherwise.